File: bindings/python/examples/faucet.py
# Copyright (c) 2025 IOTA Stiftung
# SPDX-License-Identifier: Apache-2.0

from lib.iota_sdk_ffi import *

import asyncio


async def main():
    address = Address.from_hex(
        "0x0000a4984bd495d4346fa208ddff4f5d5e5ad48c21dec631ddebc99809f16900"
    )
    faucet_client = FaucetClient.new_localnet()
    faucet_receipt = await faucet_client.request_and_wait(address)
    if faucet_receipt:
        print("Faucet receipt:")
        for coin in faucet_receipt.sent:
            print(
                f"  Coin ID: {coin.id.to_hex()}, Amount: {coin.amount}, Digest: {coin.transfer_tx_digest.to_base58()}"
            )
    else:
        print("Faucet receipt: None")


if __name__ == "__main__":
    asyncio.run(main())

File: bindings/wasm/examples/faucet.mjs
// Copyright (c) 2026 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { Address, FaucetClient, initAsync } from "@iota/sdk-wasm";

await initAsync();

const address = Address.fromHex(
  "0x0000a4984bd495d4346fa208ddff4f5d5e5ad48c21dec631ddebc99809f16900",
);
const faucetClient = FaucetClient.newLocalnet();
const faucetReceipt = await faucetClient.requestAndWait(address);
if (faucetReceipt) {
  console.log("Faucet receipt:");
  for (const coin of faucetReceipt.sent) {
    console.log(
      `  Coin ID: ${coin.id.toHex()}, Amount: ${coin.amount}, Digest: ${coin.transferTxDigest.toBase58()}`,
    );
  }
} else {
  console.log("Faucet receipt: None");
}
